### PR TITLE
Visitor can view listing

### PR DESCRIPTION
**Pull request by Carrosen**
### Changes proposed in this pull request:
* Feature test for User story
* Creates Listing model and db_columns
* Creates unit test for Listing model
* Creates FactoryBot for Listing model
* Creates Listings Controller with index view
* Sets the Listing index as a root_path
* Adds iteration on the index view in order to display listings

### What we have learned working on this feature:
* Refreshing on the HAML syntax
* Refreshing on rails console commands
* Refreshing Capybara syntax

### CODE REVIEW

**Review by leiter007 — COMMENTED** [review]
Think it looks good, cannot find anything to comment on!

File: app/views/listings/index.html.haml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+%h1 Listings#index
+- @listings.each do |listing|
+  #listing
```
Comment (tochman) [inline app/views/listings/index.html.haml]: What is the purpose of this div and how is the current indentation affecting what is being rendered by the browser?

File: features/visitor_can_view_listing.feature [hunk 2]
```diff
@@ -0,0 +1,18 @@
+Feature: Visitor can view listing
+  As a Visitor
+  In order to find a suitable host for my cat
+  I want to be able to view listings on the landing page
+
+  Background:
+    Given the following listings exist
+    | name | location   | description | email        | phone_number | availability |
+    | Zane | Gothenburg | I'm nice    | zane@cat.com | 0701234567   | 1-10 may     |
```
Comment (oliverochman) [inline features/visitor_can_view_listing.feature]: This is supposed to be an index of several listings, please add more listings

File: features/visitor_can_view_listing.feature [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+  Scenario:
+    When I visit the landing page
+    Then I should see "Zane"
+    And I should see "Gothenburg"
```
Comment (oliverochman) [inline features/visitor_can_view_listing.feature]: You need to test that you see "Gothenburg" for "Zane"

**Review by oliverochman — CHANGES_REQUESTED** [review]

File: features/visitor_can_view_listing.feature (hunk 2 above)
Comment (zanenkn) [reply features/visitor_can_view_listing.feature]: will do

File: features/visitor_can_view_listing.feature (hunk 3 above)
Comment (zanenkn) [reply features/visitor_can_view_listing.feature]: will do

Comment (tochman) [reply features/visitor_can_view_listing.feature]: Well, I would try to formulate the test in a way that calls for use of `within` in order to make sure that not only is the text visible, but it's visible on the right listing.

Comment (tochman) [reply features/visitor_can_view_listing.feature]: I guess that is what you asked for as well @oliverochman ?

Comment (oliverochman) [reply features/visitor_can_view_listing.feature]: @tochman yes :+1:

File: app/views/listings/index.html.haml [hunk 4]
```diff
@@ -0,0 +1,9 @@
+%h1 Listings#index
+- @listings.each do |listing|
+  #listing{:id=> "#{listing.id}" }
```
Comment (oliverochman) [inline app/views/listings/index.html.haml]: please use the ActiveRecord method `dom_id` to set the id here

https://apidock.com/rails/ActionView/RecordIdentifier/dom_id

**Review by oliverochman — CHANGES_REQUESTED** [review]

File: app/views/listings/index.html.haml (hunk 4 above)
Comment (tochman) [reply app/views/listings/index.html.haml]: Please use the new hash format IF you use a hash

File: app/views/listings/index.html.haml (hunk 1 above)
Comment (zanenkn) [reply app/views/listings/index.html.haml]: we will assign an id to every div (the id of the actual listing) in order to ensure and test that we have right details in the right listing (what Oliver is commenting below). This is work in progress, we have some trouble with Capibara not finding css but Oliver is looking at that.

File: features/visitor_can_view_listing.feature [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
+  Background:
+    Given the following listings exist
+    | name  | location    | description   | email         | phone_number | availability |
+    | Zane  | Gothenburg  | I'm nice      | zane@cat.com  | 0701234567   | 1-10 may     |
+    | Carla | Stockholm   | I love cats!  | carla@cool.se | 0709876543   | 25-30 may    |
+
+  Scenario:
+    When I visit the landing page
+    Then stop
```
Comment (oliverochman) [inline features/visitor_can_view_listing.feature]: you need to remove this step

File: features/visitor_can_view_listing.feature [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+    Then I should see "Zane"
+
```
Comment (oliverochman) [inline features/visitor_can_view_listing.feature]: remove empty line

**Review by oliverochman — APPROVED** [review]
Good job!

**Pull request merged**